zk_prover/src/chips/less_than_vertical.rs

```diff
@@ -0,0 +1,204 @@
+/* This is a 'vertical' implementation of LTChip
+It reduces the number of advice columns present in the original 'horizontal approach */
+
+use halo2_proofs::{
+    circuit::{Chip, Layouter, Region, Value},
+    halo2curves::{bn256::Fr as Fp, ff::PrimeField},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed, VirtualCells},
+    poly::Rotation,
+};
+
+use gadgets::{
+    bool_check,
+    util::{expr_from_bytes, pow_of_two},
+};
+
+/// Instruction that the Lt vertical chip needs to implement.
+pub trait LtVerticalInstruction {
+    /// Assign the lhs and rhs witnesses to the Lt chip's region.
+    fn assign(
+        &self,
+        region: &mut Region<'_, Fp>,
+        offset: usize,
+        lhs: Value<Fp>,
+        rhs: Value<Fp>,
+    ) -> Result<(), Error>;
+
+    /// Load the u8 lookup table.
+    fn load(&self, layouter: &mut impl Layouter<Fp>) -> Result<(), Error>;
+}
+
+/// Config for the LtVertical chip.
+#[derive(Clone, Copy, Debug)]
+pub struct LtVerticalConfig<const N_BYTES: usize> {
+    /// Denotes the lt outcome. If lhs < rhs then lt == 1, otherwise lt == 0.
+    pub lt: Column<Advice>,
+    /// Denotes the bytes representation of the difference between lhs and rhs.
+    pub diff: Column<Advice>,
+    /// Denotes the range within which each byte should lie.
+    pub u8: Column<Fixed>,
+    /// Denotes the range within which both lhs and rhs lie.
+    pub range: Fp,
+}
+
+impl<const N_BYTES: usize> LtVerticalConfig<N_BYTES> {
+    /// Returns an expression that denotes whether lhs < rhs, or not.
+    pub fn is_lt(&self, meta: &mut VirtualCells<Fp>, rotation: Option<Rotation>) -> Expression<Fp> {
+        meta.query_advice(self.lt, rotation.unwrap_or_else(Rotation::cur))
+    }
+}
+
+/// Chip that compares lhs < rhs.
+#[derive(Clone, Debug)]
+pub struct LtVerticalChip<const N_BYTES: usize> {
+    config: LtVerticalConfig<N_BYTES>,
+}
+
+impl<const N_BYTES: usize> LtVerticalChip<N_BYTES> {
+    /// Configures the LtVertical chip.
+    pub fn configure(
```

Comment:
Can you take `lt`, `diff` and `u8` as input of the `configure` function? In general we prefer to declare all the columns in the `syntesize` function of the circuit and downstream it to the chips such that we can share columns across chips